src/cruler/core.clj

```diff
@@ -199,23 +195,41 @@
      :raw-content raw-content
      :parsed-content (case file-type
                        :csv (parser/parse-csv raw-content)
-                       :text (string/split-lines raw-content)
+                       :text (parser/parse-text raw-content)
                        :yaml (parser/parse-yaml raw-content))}))
 
 (defn run-validators
-  ([validators]
-   (run-validators validators "."))
-  ([validators base-dir]
-   (reset! report-counter {:validate 0, :pass 0, :fail 0})
-   (doseq [[rule patterns] validators]
-     (log/info "\nValidating" rule)
-     (require (symbol (namespace rule)))
-     (let [data (->> (map re-pattern patterns)
-                     (filter-files base-dir)
-                     (mapv build-data1))
-           result (-> (validate rule data)
-                      (build-result rule data))]
-       (report result)))
-   (let [summary (assoc @report-counter :type :summary)]
-     (report summary)
-     summary)))
+  [validators base-dir]
+  (for [[rule patterns] validators]
+    (do
+      (require (symbol (namespace rule)))
+      (let [data (->> (map re-pattern patterns)
+                      (filter-files base-dir)
+                      (mapv build-data1))
+            result (-> (validate rule data)
+                       (build-result rule data))]
+        result))))
+
+(defn run-validators-single-file
+  [validators base-dir filepath]
```

Comment:
When we use cruler as a library, we should use like this:

```clojure
  (let [[config-file-path config] (cc/setup-config "dev-resources/sample-validator" "cruler.edn")]
    (println (cc/run-validators-single-file (:validators config) "dev-resources/sample-validator" "description/test.txt"))
```

We pass the base path to `cc/setup-config`, but this path doesn't enable in `run-validators-single-file`. 

So, when we have next directories structures:

```
dev-resources/sample-validator/
├── cruler.edn
├── description
│   └── test.txt
``` 

If the `run-validators-single-file` method doesn't have `base-dir`, we may write the next code:

```clojure
  (let [[config-file-path config] (cc/setup-config "dev-resources/sample-validator" "cruler.edn")]
    (println (cc/run-validators-single-file (:validators config) "description/test.txt"))
```

This code raised an error `file not found`, because `"description/test.txt"` means `"./description/test.txt"`. This behavior is confusing. In fact, I have made a mistake sometimes, so I decided to take a `base-dir` always.

Reply:
> This behavior is confusing. In fact, I have made a mistake sometimes, so I decided to take a base-dir always.

I see! Thx for your caring :+1: